tests/test_edgeql_ddl.py

```diff
@@ -14503,7 +14503,7 @@ async def test_edgeql_ddl_collection_cleanup_05(self):
 
         self.assertEqual(
             await self.con.query_single(count_query),
-            orig_count + 1,
+            orig_count + 2,  # one for tuple<str, str>, one for TupleExprAlias
```

Comment:
This change is actually a backporting problem, since it changes the schema for existing working cases:
```
module default {
    scalar type a extending str;
    scalar type b extending str;
    alias Bar := (<a>"", <b>"");
}
```
If you migrate to this schema on master, then apply this PR, and try to create migration again, it will fail with no proposed DDL but a schema mismatch, which is a pretty bad state.

Fortunately, `administer schema_repair();` can fix this one.

So our options are:
 * Find a way to avoid changing the behavior in cases that currently work
 * Always run a schema repair as part of the 5.4 patching
 * Figure this one will hit few enough users that we can just document `administer schema_repair();` in the changelog if they hit the problem?

Do you have thoughts on how possible the first option is?

Reply:
Option 3 is off the charts, as this will happen for anyone who has an alias or a computed global that contains a collection type. My guess would be that 25% of all edgedb databases qualify for this.

---

Old behavior:
- a single type is generated,
- TupleExprAlias with name `default::Bar`

New behavior:
- two types are generated,
- TupleExprAlias with name `default::Bar`,
- Tuple with name `tuple<default|a, default|b>` (this tuple is the material type of the TupleExprAlias)

I could make mechanism where we don't explicitly create any type that is material types of some other type.

Reply:
On the call we decided to get with option 2, because it is more correct and less likely to cause problems than option 1.